messaging/msgq.cc

```diff
@@ -82,22 +83,26 @@ void msgq_wait_for_subscriber(msgq_queue_t *q){
 }
 
 
+std::string env_or_default(const std::string& variable_name) {
+  const char* value = std::getenv(variable_name.c_str());
+  return value ? value : "DEFAULT";
+}
+
 int msgq_new_queue(msgq_queue_t * q, const char * path, size_t size){
   assert(size < 0xFFFFFFFF); // Buffer must be smaller than 2^32 bytes
   std::signal(SIGUSR2, sigusr2_handler);
 
-  const char * prefix = "/dev/shm/";
-  char * full_path = new char[strlen(path) + strlen(prefix) + 1];
-  strcpy(full_path, prefix);
-  strcat(full_path, path);
+  std::string full_path = "/dev/shm/";
+  mkdir(full_path.c_str(), 0775);
+  full_path += env_or_default("OPENPILOT_PREFIX") + "/";
+  mkdir(full_path.c_str(), 0775);
+  full_path += path;
```

Comment:
let's make the directory in the helper that setups the prefixed environment since that's what openpilot does for `/dev/shm`
```suggestion
  const char* prefix = std::getenv("OPENPILOT_PREFIX");
  if (prefix) {
    full_path += std::string(prefix);
  }
```